tests/validate-values.yaml

```diff
@@ -0,0 +1,8 @@
+---
```

Comment:
We have an issue to improve how testimony validate the tokens and its values (see https://github.com/SatelliteQE/testimony/issues/121). This is covering one of the items which is to provide a set of expected values for the token, in other words choices.

With that said I think we could start with a different yaml format that will allow us grow how we validate the values. So it would be something like:

```yaml
token:
  type: choice
  choices:
    - value1
    - value2
```

This way we can have other validations rules like:

```yaml
token:
  type: int
  min: X
  max: Y 
```

In the example above we can have an int type and then have an optional min and max value that we could validate.

Another one would be:

```yaml
token:
  type: uuid
  unique: true
```

The above would be useful to validate test IDs for example.

You don't need to care about all the other validation rules for now, implementing the choice which is the one you already have in place will be enough. I only ask if you have any suggestions for the YAML format or if you think the one I suggested is OK that you update to follow that format instead.

Reply:
@mirzal I would appreciate any comment you may have about this ^. Thank you

Reply:
@elyezer while we in Satellite don't really need other validation types, I see no problem in having forward-compatible YAML structure. So yes, I vote for supporting config like:
```yaml
token:
  type: choice
  choices:
    - choiceOne
    - choiceTwo
```

I also like the idea of using config file instead of passing tokens through command line. However, should it be more like:
```yaml
minimum tokens:
  - token1
  - token2
tokens:
  - token1
  - token2
  - token3
```
or more like:
```yaml
token:
  required: true # false is assumed by default
  type: choice
  choices:
    - choiceOne
```

This way you can just dump all expected tokens into YAML, add `required: true` to these that are considered minimal tokens, and add further validation rules to these that have special requirements. This has added benefit of ensuring that minimal tokens are subset of supported tokens, by design.

I prefer second approach, although it requires larger refactoring.

Also, can we assume token names are case-insensitive? Technically they are not, but I can't really see a reason why someone would like to have both `CaseImportance` and `caseimportance`, with two different meanings.

Reply:
@mirzal you got the idea. When I mentioned about having `required` was because I was willing to have that ditch if something is required or not so the minimum_tokens would be all the required tokens.

That said, we will be using the second format that you provided. I also think that is a good idea. Good to see that we are thinking along the same lines.

Reply:
Changed the format of the config file. Changed the way it is loaded and how actual validation is performed. Thanks.

IMO comments above needs to be part of the planned (?) issue to provide quideance.